### PR TITLE
Changes borger artifacts so you can drag people out to save them, and makes the conversion more gradual

### PR DESCRIPTION
**Pull request by frawhst**
[balance] [feature]
## About the PR 
A few changes to borger artifacts-
You can now usually (85% of the time, rolled on artifact spawn) pull people away from the artifact to interrupt the borgification process. Currently touching a borger is instant death even if you're surrounded by people, and I always found it silly that you would still get borged even if you're pulled far away by the time the process finishes. This means that lonely artifact techs can still get got, but if you touch one of these in a crowd there's a good chance someone will save you. On the other hand, the mean version now pulls you inside. It's still technically nicer than before, but you'll have to destroy the artifact to interrupt the process.

Borgers also now convert your limbs before fully borging you. Every now and again while in the artifact, you'll have one of your limbs replaced by a light robotic version, and after all the limbs are replaced, you get fully converted. This also means that people with existing robotic limbs get borged faster; after all, it's less work for the artifact. This is just a bit of flavor- getting pulled out halfway through the process will leave you partially robotized, and I think it looks cooler when people are slowly converted as opposed to the current human, human, human, GIB, robot. It would also be cool if your existing robotic limbs carried over to the cyborg you're converted into. I might add that another time, but as is you always get light parts.

The brute damage from the conversion process has been reduced, but it now makes you bleed. People who get pulled out mid-conversion will be pretty busted up. 

One performance concern:
I call update_body() on the victim every time a limb is replaced to make sure it appears immediately. That looks like a fairly expensive call, but it's called every 0.8 seconds in the worst case (for a max of 4 calls). If there's concerns about that I'm totally willing to change it.

## Why's this needed? 
Borgers being an instant death button feels dumb, and getting borged by one despite being pulled far away is also dumb. This fixes both of those at once. I don't mind if people get grabbed by one of these while alone, since that's just the dangers of artifact research, but it feels silly how there's 0 way to interrupt the conversion process even if you're right next to an artifact-poker.

The limb conversion stuff is just fun flavor. I guess you could use it to farm robot limbs if you're insane and/or have a steady supply of monkeys.

## Video
https://user-images.githubusercontent.com/67879783/120592773-3fd00a00-c3f3-11eb-8841-7202255bda45.mp4

## Changelog
```changelog
(u)Asche
(+)You can now (usually) pull people out of cyborg converter artifacts to save them. The conversion process has also been made a bit more... gradual.
```

### CODE REVIEW

**Comment by BatElite** [comment]
How long does borging take with these changes?

**Comment by frawhst** [comment]
> How long does borging take with these changes?

For a person with no robotic limbs: Average of 6.8 seconds for the inescapable version, average of 12.8 seconds for the version you can be pulled out of. Each robotic limb decreases that time by a fifth (as in, 3 robotic limbs means you'll be borged in 2/5ths of that time).

**Comment by zjdtmkhzt** [comment]
Looks pretty cool, I've been meaning to do something like this for a while! I will check out the code later, thanks! ^_^

**Comment by zjdtmkhzt** [comment]
Since the flavor text already says the device pulls the person inside and then slams shut, why not have inescapable borgers pull the person inside of themselves, like prison artifacts?

**Comment by frawhst** [comment]
> Since the flavor text already says the device pulls the person inside and then slams shut, why not have inescapable borgers pull the person inside of themselves, like prison artifacts?

good idea- will implement that in a few hours.

**Comment by Yellow-Mushroom** [comment]
Please, it’s so sad for new players and staffies to be going around and poking everything they can and then they get borged

**Comment by frawhst** [comment]
The inescapable version now pulls victims inside. Also added some bleeding damage because shit, this really should make you bleed.

File: code/obj/artifacts/artifact_objects/borg_maker.dm [hunk 1]
```diff
@@ -24,19 +32,49 @@
...
+			var/list/obj/item/parts/convertable_limbs = list(humanuser.limbs.l_arm, humanuser.limbs.r_arm, humanuser.limbs.l_leg, humanuser.limbs.r_leg)
+			//figure out which limbs are already robotic and remove them from the list
+			for (var/obj/item/parts/limb in convertable_limbs)
+				if (!limb || (limb.kind_of_limb & LIMB_ROBOT))
+					convertable_limbs -= limb
+			//people with existing robolimbs get converted faster.
+			//(loops_per_conversion_step - 1) bit adds some 'buffer time' before any limbs are converted.
+			var/loops = (loops_per_conversion_step * (convertable_limbs.len + 1)) + (loops_per_conversion_step - 1)
 			while (loops > 0)
+				if (user.loc != O.loc && user.loc != O)
```
Comment (zjdtmkhzt) [inline code/obj/artifacts/artifact_objects/borg_maker.dm:55]: Doesn't really work great with the artifact being deactivated (for instance via activator key), it just keeps going.
Maybe consider adding something like this to make it stop?
```suggestion
				if ((user.loc != O.loc && user.loc != O) || !src.activated)
```

In that case, however, the inescapable type would leave them stuck inside the artifact.
You could make more code to release everyone inside when deactivating.

People can also just damage the artifact they are inside of by flipping until it is destroyed, but I think a lot of people aren't aware of that, so it's your call if you want to leave them inside or eject them.

Apart from that minor issue, it works great in my testing!

Comment (frawhst) [reply code/obj/artifacts/artifact_objects/borg_maker.dm]: Ah, I totally forgot activator keys could do that. I think trapping people inside would create interesting situations, but you're right that not many people know about the flip-to-break thing; in fact, most people I see don't know how to break artifacts at all. For now it just ejects people on deactivation.

**Comment by frawhst** [comment]
Implemented zjd's suggestion and increased the damage of the "inescapable" version some. If you get out, you're getting out bloody as hell.

**Comment by zjdtmkhzt** [comment]
In testing I seem to have found a runtime that happens when a mob you have grabbed is qdeled or something, but that's not an issue with this PR, so, looks good!

**Pull request merged**